lib/utils/util.py

```diff
@@ -133,23 +135,31 @@ def capture_stdout_and_stderr(func, *args, **kwargs):
     return stdout_output, stderr_output
 
 
-async def capture_stdout_and_stderr_async(func, *args, **kwargs):
+async def capture_stdout_and_stderr_async(func):
+    stdout_capturer = io.StringIO()
+    stderr_capturer = io.StringIO()
+    # with contextlib.redirect_stdout(stdout_capturer):
+    with contextlib.redirect_stderr(stderr_capturer):
+        await func
+
+    stdout_capturer.flush()
+    stderr_capturer.flush()
+
+    return stdout_capturer.getvalue(), stderr_capturer.getvalue()
+
     sys.stdout.flush()
     stdout_old = sys.stdout
-    stdout_capturer = io.StringIO()
     sys.stdout = stdout_capturer
 
```

Comment:
Need the commented code? Looks like stdout isn't being captured?

Is the code after the return on line 148 reachable? If it isn't it should be removed.

Reply:
yeah, ill fix this. I was testing it out. This code is only used in tests as far as I can tell. I could not get stderr capturing working with `pytest` for some reason. I have waisted a lot of time trying to get this to work. A couple weeks back and a couple years ago I think. It works find with `unittest`

Reply:
I decided to remove this function